test/unit/ngsiv2/plugins/multientity-plugin_test.js

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2016 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-iotagent-lib
+ *
+ * fiware-iotagent-lib is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * fiware-iotagent-lib is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with fiware-iotagent-lib.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[contacto@tid.es]
+ *
+ * Modified by: Daniel Calvo - ATOS Research & Innovation
+ */
+'use strict';
+
+/* jshint camelcase: false */
+
+var iotAgentLib = require('../../../../lib/fiware-iotagent-lib'),
+    utils = require('../../../tools/utils'),
+    should = require('should'),
+    logger = require('logops'),
+    nock = require('nock'),
+    contextBrokerMock,
+    iotAgentConfig = {
+        contextBroker: {
+            host: '192.168.1.1',
+            port: '1026',
+            ngsiVersion: 'v2'
+        },
+        server: {
+            port: 4041
+        },
+        types: {
+            'WeatherStation': {
+                commands: [],
+                type: 'WeatherStation',
+                lazy: [],
+                active: [
+                    {
+                        object_id: 'p',
+                        name: 'pressure',
+                        type: 'Hgmm'
+                    },
+                    {
+                        object_id: 'h',
+                        name: 'humidity',
+                        type: 'Percentage',
+                        entity_name: 'Higro2000',
```

Comment:
entity_name and entity_type are part of the active array of attributes? The entity is associated to the device, isn't it?

(Similar comment for WeatherStation2 and WeatherStation3)

Reply:
Maybe this refers to the entity to which that attribute maps?

Reply:
Yes, this is the main functionality of the MultiEntity plugin. From https://github.com/telefonicaid/iotagent-node-lib#datamapping:

> Multientity plugin (multiEntity)
> Allows the devices provisioned in the IoTAgent to map their attributes to more than one entity, declaring the target > entity through the Configuration or Device provisioning APIs.

Reply:
NTC